tornado/iostream.py

```diff
@@ -1379,6 +1379,8 @@ def _merge_prefix(deque, size):
     >>> _merge_prefix(d, 100); print(d)
     deque(['abcdefghij'])
     """
+    if not deque:
```

Comment:
if you look at the code without knowing the issue, Isn't wrong check if the variable is None in the end and not in the beginning? I'm not an expert but this could fix it.